File: LinkedLists/LinkedList.py
class Node:
    
     def __init__(self, val):
        self.val = val
        self.next = None
        self.prev = None


class LinkedList:

    def __init__(self):
        self.head = None
        self.tail = None
    
    def addElement(self, val):

        value = Node(val)
        if self.head == None:
            self.head = value      
        else:
            curr = self.head
            while curr.next is not None:
                curr = curr.next
            curr.next = value
        
    def deleteElement(self, val):
        value = Node(val)
        curr = self.head
          
        while curr is not None:
            if curr.val == value:
                curr = curr.next
            

    
    


    
    def printList(self):
        string = ""
        curr = self.head
        while curr is not None:
            string += str(curr.val) + " --> "
            curr = curr.next
        print(string)
     

myLinkedList = LinkedList()
myLinkedList.addElement(6)
myLinkedList.addElement(7)
myLinkedList.addElement(8)
myLinkedList.addElement(9)
myLinkedList.addElement(10)
myLinkedList.printList()
print()
myLinkedList.deleteElement(8)
myLinkedList.printList()
            
            

File: LinkedLists/LinkedList.java
package LinkedLists;
import java.util.ArrayList;


public class LinkedList
{
	Node head;

	public LinkedList()
	{

	}

	public void prepend(String data)
	{
		Node newNode = new Node(data);
		newNode.next = head;
		head = newNode;

	}

	public void add(String data)
	{
		/* if its null we add it and start from there. 
 		   If not null we loop till we get null and add */

 		   Node val = new Node(data);
 		   Node current = head;

 		   if (current == null)
 		   {
 		   	head = val;

 		   } else {

 		   	while (current.next != null)
 		   	{
 		   		current = current.next;
 		   	}
 		   	current.next = val;

 		   }

	}

	public void delete(String value)
	{

		// Code on board
		for (Node current = head; current.next != null; current = current.next)
		{
			if (current.next.data.equals(value))
			{
				current.next = current.next.next;
			}
		}

		// My code
		// for (Node current = head; current != null; current = current.next)
		// {
		// 	if (current.data.equals(value))
		// 	{
		// 		current.next = current.next.next;
		// 	}
		// }
	}

	public ArrayList<String> printList()
	{
		ArrayList<String> myArray = new ArrayList<String>();

		Node current = head;

		while(current != null)
		{
			myArray.add(current.data);
			myArray.add("-->");
			current = current.next;
		}

		return myArray;
	}
       
	private class Node
	{
		String data;
		Node next;

		public Node(String data)
		{
			this.data = data;
			this.next = null;

		}

	}

}
